app/controllers/checks_controller.rb

```diff
@@ -32,6 +33,11 @@ def assessment_code
     params[:assessment_code]
   end
 
+  def clear_early_result
+    # not all checks have an early result e.g. passported clients, U18 etc
+    session_data["early_result"]&.clear
+  end
```

Comment:
So the idea here was, if a user gets to the Check Answers page, even if they have had an early (ineligible) result previously, we just want to forget about it. This is because they've given us the information for a full check, and we want to make sure we treat their check, even if they go back and forth and do weird stuff, as a full check. This is to try and avoid any complexity associated with a previous partial check.

In relation to this bullet, `clear_early_result will trigger when early_result key exists and is nil/null` - def clear_early_result will clear the early result information in the session data, if it exists. If it doesn't exist, it'll just quietly fail as we know there are some situations where it won't exist.

Currently none of this new code triggers metrics tracking - my initial thoughts on this were, kick off an analytics event when we get the early ineligible result (probably inside def calculate_early_result or somewhere like that) - this event will save the current session data in the JSONB field. 

If they do a full check, I think it's okay to remove the early result from the session data as we have other ways to link a full check to an early ineligible check (assessment_id). Does that make sense?

Reply:
Ok, sounds good. 

I need to think about this properly. I'm struggling to work out why we want to ever forget the information  in `early_result` when the user does a full check. This is a 'me' problem not the 'PR' problem. 😅

Reply:
No that's okay! I did um and ah about adding this in. But ultimately I'm trying to protect against weird user scenarios where they go back and forth like:
- user gets early ineligible result
- goes to results page
- goes 'back'
- decides to continue to enter all data
- gets to Check Answers
- submits

in this scenario they would see the reduced version of the results page, where capital, disposable aren't rendered. But they would have entered the info. It felt like just clearing the early result stuff on Check Answers was the simplest way to avoid weird journeys like this.

Reply:
(this is the problem with setting stuff in session_data - you have to clean up after yourself properly or get weird behaviour)